### PR TITLE
add option for explicit pressure for rock compaction (only transmissibility)

### PR DESCRIPTION
**Pull request by totto82**
The option is false per default. i.e we use implicit as default.

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this please

File: opm/simulators/wells/WellInterface_impl.hpp [hunk 1]
```diff
@@ -1562,7 +1562,7 @@ namespace Opm
         for (int perf = 0; perf < this->number_of_perforations_; ++perf) {
             const int cell_idx = this->well_cells_[perf];
             const auto& intQuants = simulator.model().intensiveQuantities(cell_idx, /*timeIdx=*/ 0);
-            const double trans_mult = simulator.problem().template rockCompTransMultiplier<double>(intQuants, cell_idx);
+            const double trans_mult = simulator.problem().template wellTransMultiplier<double>(intQuants, cell_idx);
```
Comment (totto82) [inline opm/simulators/wells/WellInterface_impl.hpp:1565]: Here I include the perm reduction due to salt contribution to the output of the connection trans factor. The test failure are from this change. @plgbrts Is this inline with what you think should be outputted?

File: ebos/eclproblem.hh [hunk 2]
```diff
@@ -1657,37 +1661,11 @@ public:
...
-            return this->rockCompTransMult_[tableIdx].eval(effectiveOilPressure, /*extrapolation=*/true);
-
-        // water compaction
-        assert(!this->rockCompTransMultWc_.empty());
-        LhsEval SwMax = max(decay<LhsEval>(fs.saturation(waterPhaseIdx)), this->maxWaterSaturation_[elementIdx]);
-        LhsEval SwDeltaMax = SwMax - initialFluidStates_[elementIdx].saturation(waterPhaseIdx);
-
-        return this->rockCompTransMultWc_[tableIdx].eval(effectiveOilPressure, SwDeltaMax, /*extrapolation=*/true);
+        bool implicit = !EWOMS_GET_PARAM(TypeTag, bool, ExplicitRockCompaction);
+        return implicit? this->simulator().problem().template computeRockCompTransMultiplier_<LhsEval>(intQuants, elementIdx) :
```
Comment (atgeirr) [inline ebos/eclproblem.hh:1665]: Formatting: space before `?`, put the ':' on the next line, aligned with the '?'. Same on line 1698.

Extra whitespace on line 1668.

File: ebos/eclproblem.hh [hunk 3]
```diff
@@ -1716,7 +1694,9 @@ public:
     {
         OPM_TIMEBLOCK_LOCAL(wellTransMultiplier);
         
-        double trans_mult = this->simulator().problem().template rockCompTransMultiplier<double>(intQuants, elementIdx);
+        bool implicit = !EWOMS_GET_PARAM(TypeTag, bool, ExplicitRockCompaction);
+        double trans_mult = implicit? this->simulator().problem().template computeRockCompTransMultiplier_<double>(intQuants, elementIdx) :
+                            this->simulator().problem().getRockCompTransMultVal(elementIdx);
         trans_mult *= this->simulator().problem().template permFactTransMultiplier<double>(intQuants);
```
Comment (atgeirr) [inline ebos/eclproblem.hh:1700]: This is still implicit, should this factor also be explicit if RockCompaction is?

If keeping as is, you could simplify with
```
double trans_mult = rockCompTransMultiplier<double>(intQuants, elementIdx);
```
which already contain the above logic.

**Review by atgeirr — COMMENTED** [review]
Looks good, just have a question about the perm factor being explicit and a formatting nitpick.

File: ebos/eclproblem.hh (hunk 3 above)
Comment (totto82) [reply ebos/eclproblem.hh:1700]: This is still implicit, should this factor also be explicit if RockCompaction is?

I dont think that is necessary.  The perfFactTransMultiplier is related to salt precipitation and are different from rock compaction.

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
jenkins build this update_data please

**Comment by totto82** [comment]
jenkins build this opm-tests=1119 please

**Pull request merged**